From 3cdccb2b601c27d0540db6379a5a13a30c8b4db6 Mon Sep 17 00:00:00 2001
From: Sergii Tkachenko <sergiitk@google.com>
Date: Tue, 23 Dec 2025 13:17:24 -0800
Subject: [PATCH 1/2] [Python][Bazel] bump rules_python to 1.6.3 (#41278)

rules_python changelog:
- https://rules-python.readthedocs.io/en/latest/changelog.html#v1-6-0
- https://rules-python.readthedocs.io/en/latest/changelog.html#v1-6-3

Notably, this fixes [rules_python repl](https://rules-python.readthedocs.io/en/latest/repl.html):

```
$ bazel --quiet run @rules_python//python/bin:repl
INFO: Running bazel wrapper (see //tools/bazel for details), bazel version 8.0.1 will be used instead of system-wide bazel installation.
Target @@rules_python//python/bin:repl up-to-date:
  bazel-bin/external/rules_python/python/bin/repl
  bazel-bin/external/rules_python/python/bin/repl_py.py
Python 3.13.11 (main, Dec  6 2025, 02:15:39) [Clang 17.0.0 (clang-1700.3.19.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>>
```

This was previously broken in 1.5.4.

Closes #41278

COPYBARA_INTEGRATE_REVIEW=https://github.com/grpc/grpc/pull/41278 from sergiitk:py/bazel/rules_python 1e4f692f99915ce5fe7fcf0efd88ace22198d889
PiperOrigin-RevId: 848277600
---
 MODULE.bazel                | 4 ++--
 bazel/grpc_python_deps.bzl  | 6 +++---
 templates/MODULE.bazel.inja | 4 ++--
 3 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/MODULE.bazel b/MODULE.bazel
index 2a1c5aec1e..fe0c8d5f73 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -60,7 +60,7 @@ bazel_dep(name = "rules_shell", version = "0.4.0", dev_dependency = True)
 # Python dependencies
 # ===================
 
-bazel_dep(name = "rules_python", version = "1.5.4")
+bazel_dep(name = "rules_python", version = "1.6.3")
 
 PYTHON_VERSIONS = [
     "3.9",
@@ -68,7 +68,7 @@ PYTHON_VERSIONS = [
     "3.11",
     "3.12",
     "3.13",
-    "3.14.0b2",
+    "3.14",
 ]
 
 python = use_extension("@rules_python//python/extensions:python.bzl", "python")
diff --git a/bazel/grpc_python_deps.bzl b/bazel/grpc_python_deps.bzl
index 4c3def0332..a07079057c 100644
--- a/bazel/grpc_python_deps.bzl
+++ b/bazel/grpc_python_deps.bzl
@@ -22,9 +22,9 @@ def grpc_python_deps():
     if "rules_python" not in native.existing_rules():
         http_archive(
             name = "rules_python",
-            sha256 = "13671d304cfe43350302213a60d93a5fc0b763b0a6de17397e3e239253b61b73",
-            strip_prefix = "rules_python-1.5.4",
-            url = "https://github.com/bazel-contrib/rules_python/releases/download/1.5.4/rules_python-1.5.4.tar.gz",
+            sha256 = "2f5c284fbb4e86045c2632d3573fc006facbca5d1fa02976e89dc0cd5488b590",
+            strip_prefix = "rules_python-1.6.3",
+            url = "https://github.com/bazel-contrib/rules_python/releases/download/1.6.3/rules_python-1.6.3.tar.gz",
         )
 
     python_configure(name = "local_config_python")
diff --git a/templates/MODULE.bazel.inja b/templates/MODULE.bazel.inja
index 53108d9382..e2b88a77b7 100644
--- a/templates/MODULE.bazel.inja
+++ b/templates/MODULE.bazel.inja
@@ -60,7 +60,7 @@ bazel_dep(name = "rules_shell", version = "0.4.0", dev_dependency = True)
 # Python dependencies
 # ===================
 
-bazel_dep(name = "rules_python", version = "1.5.4")
+bazel_dep(name = "rules_python", version = "1.6.3")
 
 PYTHON_VERSIONS = [
     "3.9",
@@ -68,7 +68,7 @@ PYTHON_VERSIONS = [
     "3.11",
     "3.12",
     "3.13",
-    "3.14.0b2",
+    "3.14",
 ]
 
 python = use_extension("@rules_python//python/extensions:python.bzl", "python")
-- 
2.53.0.273.g2a3d683680-goog

